crypto/src/asset.rs

```diff
@@ -28,8 +34,62 @@ use crate::Fq;
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub struct Id(pub Fq);
 
+/// Validator-specific unique assets, used for identifying staking tokens
+/// associated with a particular validator.
+pub struct ValidatorAsset {
+    validator_pubkey: PublicKey,
+}
+
+impl ValidatorAsset {
+    pub fn new(pk: PublicKey) -> Self {
+        ValidatorAsset {
+            validator_pubkey: pk,
+        }
+    }
+
+    pub fn denom(&self) -> Denom {
+        Denom(format!(
+            "d{}-{}",
+            NATIVE_ASSET_DENOM,
+            self.validator_pubkey
+                .to_bech32(PENUMBRA_BECH32_VALIDATOR_PREFIX)
+        ))
+    }
+
+    pub fn id(&self) -> Id {
+        Id::from(self.denom())
+    }
+}
+
+pub fn is_validator_denom(d: &Denom) -> bool {
+    let re = Regex::new(&format!(
+        r"^d{}-({}{}\w+)$",
+        NATIVE_ASSET_DENOM,
+        PENUMBRA_BECH32_VALIDATOR_PREFIX,
+        // Note, if tendermint-rs ever changes the bech32 separator from default,
+        // this will break
+        bech32::DEFAULT_SEPARATOR
+    ))
+    .unwrap();
+    let caps = re.captures(&d.0);
+    if caps.is_none() {
+        // There was no bech32 address captured
+        return false;
+    }
+
+    let bech32_address = caps.unwrap().get(1).map_or("", |m| m.as_str());
+
+    re.is_match(&d.0) && bech32::decode(bech32_address.to_lowercase()).is_ok()
+}
+
 pub struct Denom(pub String);
 
+impl std::fmt::Display for Denom {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.0)
```

Comment:
Ideally `Display` and `FromStr` roundtrip, which I think would happen if we display the associated `Denom`:

```suggestion
        write!(f, "{}", self.denom().0)
```